Controller/Process/Result.php

```diff
@@ -515,6 +527,7 @@ protected function validatePayloadAndReturnResponse($result)
 
         try {
             $response = $service->paymentsDetails($request);
+            $this->paymentResponseHandler->handlePaymentResponse($response, $this->payment, $order);
```

Comment:
This controller already has a method `_validateUpdateOrder` to handle the payments response. I think that bringing the whole logic of `handlePaymentResponse` here might not be good. The alternative solution would require some code duplication but since it is a different flow, it could be necessary.